lib/ansible/plugins/inventory/constructed.py

```diff
@@ -116,6 +116,11 @@ def parse(self, inventory, loader, path, cache=False):
                 # create composite vars
                 self._set_composite_vars(data.get('compose'), hostvars, host, strict=strict)
 
+                # refetch host vars in case new ones have been created above
```

Comment:
we could just have _set_composite_vars return the new vars

Reply:
Would this be the preference? Alternatively I considered _set_composite_vars returning true/false to indicate whether host vars need updating.